src/makeCalculator.js

```diff
@@ -37,7 +37,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(num) {
+      calculator.result += num;
+
+      return calculator;
+    },
+
+    subtract(num) {
+      calculator.result -= num;
+
+      return calculator;
+    },
+
+    multiply(num) {
+      calculator.result *= num;
+
+      return calculator;
+    },
+
+    divide(num) {
+      calculator.result /= num;
+
+      return calculator;
+    },
+
+    reset(num) {
+      calculator.result = 0;
+
+      return calculator;
+    },
+
+    operate(operation, num) {
+      return operation(num);
+    },
```

Comment:
operation will lose `this`. it can be reassign with call method
```suggestion
    operate(operation, num) {
      operation.call(this, num);
      
      return this;
    },
```